sonar-php-plugin/src/main/java/org/sonar/plugins/php/SymbolScanner.java

```diff
@@ -45,12 +45,14 @@ public class SymbolScanner extends Scanner {
   private final ActionParser<Tree> parser = PHPParserBuilder.createParser();
   private final ProjectSymbolData projectSymbolData = new ProjectSymbolData();
   private final Cache cache;
+  private final CacheContextImpl cacheContext;
 
   private int symbolTablesFromCache = 0;
 
-  SymbolScanner(SensorContext context, DurationStatistics statistics, Cache cache) {
+  SymbolScanner(SensorContext context, DurationStatistics statistics) {
     super(context, statistics);
-    this.cache = cache;
+    this.cacheContext = CacheContextImpl.of(context);
+    this.cache = new Cache(cacheContext);
```

Comment:
Personally, I don't like any initialization inside of the constructor (outside calling super() or this() or initializing fields), because it makes unit testing harder. Consider having this outside constructor if possible.